pkg/objstore/objstore.go

```diff
@@ -412,7 +412,7 @@ func (b *metricBucket) Delete(ctx context.Context, name string) error {
 
 	start := time.Now()
 	if err := b.bkt.Delete(ctx, name); err != nil {
-		if !b.isOpFailureExpected(err) {
+		if !b.isOpFailureExpected(err) && ctx.Err() == nil {
```

Comment:
```suggestion
		if !b.isOpFailureExpected(err) && ctx.Err() != context.Canceled {
```

Reply:
Agree timeouts are legit failures, but this comment apply to any place, not just here, right? Also, is there a good reason why we don't we the `context.Canceled` check `isOpFailureExpected()`?

Reply:
From what I have observed, cancelled requests can cause timeouts:
<img width="1838" alt="Screen Shot 2020-09-10 at 10 43 58 AM" src="https://user-images.githubusercontent.com/332617/93372963-a63fd880-f822-11ea-8057-8992e7042555.png">

If you notice, query cancels the request but store reports it as aborted. In my opinion that should be reported as cancelled, and not increase the error count.

Reply:
or does Aborted != timeout?

Reply:
> Agree timeouts are legit failures, but this comment apply to any place, not just here, right? Also, is there a good reason why we don't we the `context.Canceled` check `isOpFailureExpected()`?

If that is a better place for this then I have no problems making the change.

Reply:
`isOpFailureExpected` is really only for user specified function what to expect. I believe Cancel is valid to be ignored no matter what user specifies, no? (:

Reply:
You're right @bwplotka 👍

Reply:
I've updated the PR to use that context.